Support for RSA-PKCS1v15-HASH_ID scheme

danixeee opened the pull request:
Add support for `RSA-PKCS1v15` signatures, as well as other supported openssl hash algorithms.

**Description of the changes being introduced by the pull request**:

Allow passing one of the following schemes, when generating keys and creating/verifying signatures:

- `rsassa-pss-HASH_ID`
- `rsa-pkcs1v15-HASH_ID`

Where `HASH_ID` is one of: `md5`, `sha1`, `sha224`, `sha256`, `sha384` and `sha512`.

`HASH_ID` is picked from a _scheme_, so changes are backwards compatible.

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

danixeee commented:
Hi @lukpueh, I didn't see that you are working on `gpg` transition on your fork. I was looking at branches that belongs to this repository, but now I can see that our work does not overlap, which is great. :)

I guess we should continue the conversation from #163 here.

> Note that I didn't adapt SIGNATURE_SCHEMA in #174 as you did here, because it felt like a bigger change. I'd rather do this in a separate PR, maybe as part of a broader formats/schema revision. Do you think you can work around this by adding something like below to your taf.formats and patching append_signature accordingly?

Do you think I should merge your work (#174) to mine (#173) and update things that you have proposed in this PR?

lukpueh commented:
> Do you think I should merge your work (#174) to mine (#173) and update things that you have proposed in this PR?

No need to do that. Let's treat them separately while reviewing, and rebase the pending one, once the other one gets merged. I suspect your's will be merged sooner than mine, so I'll do the rebase on my side. :)

danixeee commented on securesystemslib/keys.py:
```diff
@@ -121,6 +121,20 @@
+  'rsa-pkcs1v15-sha256',
+  'rsa-pkcs1v15-sha512',
+]
+
```
I left it accidentally, thanks.

danixeee commented on securesystemslib/hash.py:
```diff
@@ -47,6 +47,82 @@
+
+  # Dictionary of `pyca/cryptography` supported hash algorithms.
+  PYCA_DIGEST_OBJECTS_CACHE = {hash_algo.name: hash_algo
+                               for hash_algo in cryptography.hazmat.primitives.hashes.HashAlgorithm._abc_registry}
```
Agreed, I think that explicit way is the most readable. I had similar code already (https://github.com/openlawlibrary/securesystemslib/blob/e5051e610a6e6d3c58c9b72d3d79e1ed955ea188/securesystemslib/pyca_crypto_keys.py#L159), not sure why I changed it. :)

danixeee commented:
@lukpueh I think that I have addressed all your comments for now, thank you for the fast review!

> Would you please revert the alignments of continued lines in pyca_crypto_keys.py that are unrelated to this PR (was that your IDE?)?

It was, I am using VS Code but I disabled all formatters, not sure why it changed those alignments. It should be fine now.

> Do you think I should merge your work (#174) to mine (#173) and update things that you have proposed in this PR?

I asked this because I will need _gpg_ formats from your PR in order to change schemas. Should I copy them manually then?

lukpueh commented:
> I asked this because I will need gpg formats from your PR in order to change schemas. Should I copy them manually then?

I'll merge this PR now and will rebase my `lukpueh:add-gpg` from #174 onto master. I suggest you work on top of `lukpueh:add-gpg` then. It shouldn't change a lot, because the logic has already been reviewed in in-toto. However, it still needs someone to take a look at it before we merge it here (cc @SantiagoTorres, @mnm678).
